src/main/resources/lib/translate/queue.ts

```diff
@@ -84,23 +113,27 @@ class TaskQueue {
             callback: () => {
                 logDebug(
                     LogDebugGroups.CRON,
-                    `Active tasks: ${this.activeTasks.size}, queue length: ${this.queue.length}`,
+                    `Active tasks: ${this.activeTasks.size}, queue length: ${this.queuedTasksGroups.length}`,
                 );
 
-                this.activeTasks.forEach((task, taskId) => {
-                    const state = taskLib.get(taskId)?.state ?? 'FINISHED';
-                    switch (state) {
-                        case 'FINISHED':
-                            this.activeTasks.delete(taskId);
-                            break;
-                        case 'FAILED':
-                            this.activeTasks.delete(taskId);
-                            task.onError?.();
-                            break;
-                    }
-                });
-
-                this.runNextTasks();
+                this.synchronizer.sync(
+                    __.toScriptValue(() => {
```

Comment:
Do we need to create a new instance of this bean per `TaskQueue`? If not, I may be better moving it to a separate utility file and export something like `runSync(fn)`, that will wrap the call into `sync` and `__.toScriptValue`